src/main/java/org/fenixedu/academic/ui/spring/AcademicBaseController.java

```diff
@@ -0,0 +1,39 @@
+package org.fenixedu.academic.ui.spring;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+
+public class AcademicBaseController {
+    //The HTTP Request that can be used internally in the controller
+    protected @Autowired HttpServletRequest request;
```

Comment:
Good Spring practices dictate that access to the raw request should be kept to a minimum, so it is bad practice to expose it by default in every situation.